lib/resourcebuilder/apps.go

```diff
@@ -56,6 +58,7 @@ func (b *deploymentBuilder) Do(ctx context.Context) error {
 func waitForDeploymentCompletion(ctx context.Context, client appsclientv1.DeploymentsGetter, deployment *appsv1.Deployment) error {
 	return wait.PollImmediateUntil(defaultObjectPollInterval, func() (bool, error) {
 		d, err := client.Deployments(deployment.Namespace).Get(deployment.Name, metav1.GetOptions{})
+		deploymentConditions := make([]string, 0, len(d.Status.Conditions))
```

Comment:
d can be nil when err is not nil... move this to https://github.com/openshift/cluster-version-operator/pull/187/files#diff-20dbece9affd4bfefb9de7caca6bc037R80

Reply:
cc @jcpowermac

Reply:
@abhinavdahiya thanks! fixed